python/ray/tests/test_cli.py

```diff
@@ -978,6 +978,82 @@ def output_ready():
         _check_output_via_pattern("test_ray_status_multinode_v1.txt", result)
 
 
+def test_ray_drain_node(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(num_cpus=1, resources={"node1": 1})
+    ray.init(address=cluster.address)
+    cluster.add_node(num_cpus=1, resources={"node2": 1})
+    cluster.add_node(num_cpus=1, resources={"node3": 1})
```

Comment:
there's no reason to do multi-node integration tests here, just use a fake GCS client and ensure the requests are made properly

please also add tests for validation (bad timestamps, bad enums)